### PR TITLE
Bug-fix/find_orderings_of_exprs

### LINKED ISSUE #7418: `panicked at 'index out of bounds: the len is 0 but the index is 0'` in `find_orderings_of_exprs`
### Describe the bug

While working on upgrading DataFusion in IOx in https://github.com/influxdata/influxdb_iox/pull/8577 one of our integration tests is failing like this:
```
---- physical_optimizer::projection_pushdown::tests::test_integration stdout ----
thread 'physical_optimizer::projection_pushdown::tests::test_integration' panicked at 'index out of bounds: the len is 0 but the index is 0', /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/aa1d72c/datafusion/physical-expr/src/utils.rs:954:54
stack backtrace:
   0: rust_begin_unwind
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/panicking.rs:67:14
   2: core::panicking::panic_bounds_check
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/panicking.rs:162:5
   3: datafusion_physical_expr::utils::find_orderings_of_exprs::{{closure}}
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/aa1d72c/datafusion/physical-expr/src/utils.rs:954:54
   4: core::option::Option<T>::map
             at /rustc/eb26296b556cef10fb713a38f3d16b9886080f26/library/core/src/option.rs:1075:29
   5: datafusion_physical_expr::utils::find_orderings_of_exprs
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/aa1d72c/datafusion/physical-expr/src/utils.rs:954:9
   6: datafusion::physical_plan::projection::ProjectionExec::try_new
             at /Users/alamb/.cargo/git/checkouts/arrow-datafusion-71ae82d9dec9a01c/aa1d72c/datafusion/core/src/physical_plan/projection.rs:143:25
   7: iox_query::physical_optimizer::projection_pushdown::wrap_user_into_projections
             at ./src/physical_optimizer/projection_pushdown.rs:398:25
   8: <iox_query::physical_optimizer::projection_pushdown::ProjectionPushdown as datafusion::physical_optimizer::optimizer::PhysicalOptimizerRule>::optimize::{{closure}}
```

The test is here: https://github.com/influxdata/influxdb_iox/blob/2eb74ddb87bb1da2899518a095854c150b21ef03/iox_query/src/physical_optimizer/projection_pushdown.rs#L1318

I believe it is failing in `find_orderings_of_exprs` that was added in https://github.com/apache/arrow-datafusion/pull/7364 by @berkaysynnada (looks like it needs a check for empty).

### To Reproduce

See https://github.com/influxdata/influxdb_iox/pull/8577

### Expected behavior

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by berkaysynnada**
## Which issue does this PR close?

Closes #7418 .

## Rationale for this change

IOx tests failed. This error has not been encountered in Datafusion because in the absence of output ordering, `input.output_ordering()` returns None, not Some([]). This issue is fixed now.

### CODE REVIEW

**Comment by liukun4515** [comment]
@berkaysynnada can you add a ut for that case?

**Review by ozankabak — APPROVED** [review]
LGTM, a fairly small and straightforward bugfix. Let's merge soon if there are no concerns.

File: datafusion/physical-expr/src/utils.rs [hunk 1]
```diff
@@ -1972,4 +1977,55 @@ mod tests {
...
+
+        assert_eq!(
+            vec![
+                Some(PhysicalSortExpr {
+                    expr: Arc::new(Column::new("b_new", 0)),
+                    options: SortOptions::default(),
+                }),
+                None,
+            ],
+            empty_ordering
```
Comment (viirya) [inline datafusion/physical-expr/src/utils.rs:2009]: nit: it doesn't look like empty.

```suggestion
        let one_ordering = find_orderings_of_exprs(
            &[
                (Arc::new(Column::new("b", 1)), "b_new".to_string()),
                (Arc::new(Column::new("a", 0)), "a_new".to_string()),
            ],
            Some(&[PhysicalSortExpr {
                expr: Arc::new(Column::new("b", 1)),
                options: SortOptions::default(),
            }]),
            EquivalenceProperties::new(Arc::new(schema.clone())),
            OrderingEquivalenceProperties::new(Arc::new(schema.clone())),
        )?;

        assert_eq!(
            vec![
                Some(PhysicalSortExpr {
                    expr: Arc::new(Column::new("b_new", 0)),
                    options: SortOptions::default(),
                }),
                None,
            ],
            one_ordering
```

**Review by viirya — APPROVED** [review]

**Comment by alamb** [comment]
Thank you @berkaysynnada and @viirya and @ozankabak

**Pull request merged**